Dockerfile

```diff
@@ -1,19 +1,35 @@
-# a big docker file meant for depolying min-turnup
 FROM ubuntu
 
-RUN apt-get update
-RUN apt-get install -y curl vim jq libssl-dev openssl zip make sudo
-
-RUN curl -sSL --fail -o /usr/local/bin/jsonnet \
-  "https://storage.googleapis.com/kube-deploy/jsonnet/adf169b1e4a4ba170f58b47111ed88552fae42c8"
-RUN chmod +x /usr/local/bin/jsonnet
+# Install basics
+RUN apt-get update && apt-get -y upgrade \
+    && apt-get install -y git curl vim jq libssl-dev openssl zip make sudo \
+    && rm -rf /var/lib/apt/lists/*
 
+# Install Terraform
+ENV TERRAFORM_VERSION 0.7.0-rc1
 RUN curl -sSL --fail \
-  "https://releases.hashicorp.com/terraform/0.6.16/terraform_0.6.16_linux_amd64.zip" \
-  -o /tmp/tf.zip \
-  && unzip -d /usr/local/bin /tmp/tf.zip \
-  && rm /tmp/tf.zip
+    "https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/terraform_${TERRAFORM_VERSION}_linux_amd64.zip" \
+    -o /tmp/tf.zip \
+    && unzip -d /usr/local/bin /tmp/tf.zip \
+    && rm /tmp/tf.zip
+
+# Azure - Required dependencies 
+RUN apt-get update && apt-get -y upgrade && \
+    apt-get install -y nodejs npm && \
+    rm -rf /var/lib/apt/lists/*
+RUN npm install -g azure-cli
```

Comment:
Terraform Azure doesn't require the CLI, but it requires a ServicePrincipal "service account" which isn't super obvious or easy to create on Azure.

More importantly, the CloudProvider require a ServicePrincipal with certain permissions to talk to Azure APIs. There is no VM Identity as there is in GCE/AWS so this has to be done "manually". I was going to extend the Azure phase1 to offer the create the SP for the user with permissions scoped to the ResourceGroup the cluster is being deployed into. (RGs create boundaries for easy cleanup and permissions).

Currently, the [preliminary instructions I have for kubernetes-anywhere on Azure](https://github.com/colemickens/azure-kubernetes-demo) asks the user to bring their own ServicePrincipal but that's inconvenient for a number of reasons. 

I was including the Azure CLI here in order to extend Azure's phase1 offer to create the ServicePrincipal on the users' behalf. I can remove it for now though, it's not used yet and we can discuss this aspect at a later time.